### PR TITLE
feat: LT-1522 - Exporter Generated Docs

### PR DESCRIPTION
**Pull request by andytaylorkainos**

### CODE REVIEW

File: applications/views/common.py [hunk 1]
```diff
@@ -152,6 +153,13 @@ def get(self, request, **kwargs):
         ecju_query_notifications = len(
             [x for x in notifications if x["parent"] == self.application_id and x["object_type"] == "ecju_query"]
         )
+        generated_case_document_notifications = len(
+            [
+                x
```
Comment (BenAndrew-Kainos) [inline applications/views/common.py]: Please give a better name than x

File: applications/views/common.py [hunk 2]
```diff
@@ -152,6 +153,13 @@ def get(self, request, **kwargs):
...
+                for x in notifications
+                if x["parent"] == self.application_id and x["object_type"] == "generated_case_document"
```
Comment (BenAndrew-Kainos) [inline applications/views/common.py]: Please define "generated_case_document" as a constant or in an enum somewhere

File: ui_automation_tests/features/generated_documents.feature [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+Feature: As a logged in exporter
+I want to see when there are Generated Documents relating to my applications
+So that I can download the documentation
+
+  @LT_1552_generated_documents
+  Scenario: view Generated Documents on an application
+    Given I go to exporter homepage and choose Test Org
+    When I go to the recently created application with a Generated Document attached
+    And I click the Generated Documents tab
+    Then I can see the Generated Document
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/features/generated_documents.feature]: newline

File: applications/urls.py [hunk 4]
```diff
@@ -186,6 +186,12 @@
         name="delete_additional_document",
     ),
     path("<uuid:pk>/withdraw/", common.WithdrawApplication.as_view(), name="withdraw"),
+    # Case-relevant documentation
+    path(
+        "<uuid:pk>/generated-documents/<uuid:obj_pk>/download",
```
Comment (Mark-Spiers-kainos) [inline applications/urls.py]: could you change this to doc_pk?

File: applications/views/common.py (hunk 1 above)
Comment (andytaylorkainos) [reply applications/views/common.py]: done

File: applications/urls.py (hunk 4 above)
Comment (andytaylorkainos) [reply applications/urls.py]: discusssed in person as to why i cant

File: ui_automation_tests/features/generated_documents.feature [hunk 5]
```diff
@@ -0,0 +1,11 @@
+@generated_documents @all
+Feature: As a logged in exporter
+I want to see when there are Generated Documents relating to my applications
+So that I can download the documentation
+
+  @LT_1552_generated_documents
```
Comment (phlptt) [inline ui_automation_tests/features/generated_documents.feature]: can you add smoke or regression to this please
@smoke - for quick runs making sure core functionality is running (slow)

@regression - everything else (fast)

Comment (andytaylorkainos) [reply ui_automation_tests/features/generated_documents.feature]: sure!

File: applications/services.py [hunk 6]
```diff
@@ -281,6 +282,16 @@ def get_application_ecju_queries(request, pk):
     return open_queries, closed_queries
 
 
+def get_application_generated_documents(request, pk):
+    data = get(request, APPLICATIONS_URL + pk + GENERATED_DOCUMENTS_URL).json()["generated_documents"]
+    return data
+
+
+def get_generated_document(request, pk, doc_pk):
+    data = get(request, APPLICATIONS_URL + pk + GENERATED_DOCUMENTS_URL + str(doc_pk) + "/")
```
Comment (Mark-Spiers-kainos) [inline applications/services.py]: can doc_pk be passed in as a str, rather than casted in the function, to keep it consistent?

Comment (andytaylorkainos) [reply applications/services.py]: Discussed in person

**Pull request merged**